File: 品味Java的21種設計模式/dp_java/bridge/example7/AbstractMessage3.java
package cn.javass.dp.bridge.example7;

/**
 * 抽象的消息对象
 */
public abstract class AbstractMessage3 {
    /**
     * 持有一个实现部分的对象
     */
    protected MessageImplementor impl;
    /**
     * 构造方法
     */
    public AbstractMessage3() {
        //创建一个默认的实现
        this.impl = new MessageSMS();
    }

    /**
     * 发送消息，转调实现部分的方法
     * @param message 要发送的消息内容
     * @param toUser 把消息发送的目的人员
     */
    public void sendMessage(String message, String toUser) {
        this.impl.send(message, toUser);
    }

}

File: 品味Java的21種設計模式/dp_cpp/bridge/example7/AbstractMessage3.cpp
#include "AbstractMessage3.h"

namespace cn {
namespace javass {
namespace dp {
namespace bridge {
namespace example7 {

AbstractMessage3::AbstractMessage3() {
    //创建一个默认的实现
    this->impl = new MessageSMS();
}

void AbstractMessage3::sendMessage(std::string message, std::string toUser) {
    this->impl->send(message, toUser);
}
}
}
}
}
}
